include/picongpu/fields/currentDeposition/EmZ/DepositCurrent.hpp

```diff
@@ -80,18 +80,31 @@ namespace picongpu
                 {
                     if(line.m_pos0[2] == line.m_pos1[2])
                         return;
+
+                    auto const shape_i = make_TrajectoryAssignmentShapeFunction(
+                        typename T_Strategy::template Shape_i<ParticleAssign>(line.m_pos0[0], true),
+                        typename T_Strategy::template Shape_i<ParticleAssign>(line.m_pos1[0], true));
+
+                    auto const shape_j = make_TrajectoryAssignmentShapeFunction(
+                        typename T_Strategy::template Shape_j<ParticleAssign>(line.m_pos0[1], true),
+                        typename T_Strategy::template Shape_j<ParticleAssign>(line.m_pos1[1], true));
```

Comment:
This looks very hard to read, and also repetitive. I think it can be wrapped in some helper function to better convay what is happening and also avoid repeating an almost similar line 6 times. Or at least please use `{}` for initialization to distinguish from a function call `()`, as here it is really easy to get confused with that `::template`